gbpservice/neutron/services/servicechain/drivers/oneconvergence_intercloud_driver.py

```diff
@@ -141,63 +160,97 @@ def _fetch_template_and_params(self, context, sc_instance,
                 external_net_id = ext_networks[0]['id']
                 config_param_values['external_network_id'] = external_net_id
 
-        svc_mgmt_pt = self.create_pt(context, svc_mgmt_ptgs[0]['id'])
-
-        filters = {'id': [consumer_ptg_id]}
-        ptgs = self._grouppolicy_plugin.get_policy_target_groups(
-                    context._plugin_context, filters)
-        consumer_ptg_subnet_id = None
-        stitching_port_ip = None
-        if ptgs:
-            is_consumer_ptg_external = False
-            consumer_ptg_subnet_id = self._get_ptg_subnet(
-                                        context, consumer_ptg_id)
-        else:
-            is_consumer_ptg_external = True
+        consumer_port_id = None
+        stitching_port_id = None
+
+        admin_context._plugin_context = copy.copy(admin_context)
+        provider_ptg_subnet_id = self._get_ptg_subnet(context, provider_ptg_id)
+        svc_mgmt_ptg_subnet_id = self._get_ptg_subnet(admin_context, svc_mgmt_ptgs[0]['id'])
+        mgmt_subnet = self._core_plugin.get_subnet(admin_context, svc_mgmt_ptg_subnet_id)
+        svc_mgmt_network_id = mgmt_subnet['network_id']
+
+        admin_tenant = simplechain_driver.get_admin_tenant_object()
+        admin_context.tenant_name = admin_tenant.name
+        admin_context.tenant_id = admin_tenant.id
+
+        port_name = "mgmt-port"
+        attrs = {'port': {'tenant_id': admin_context.tenant_id,
+                 'name': port_name,
+                 'network_id': svc_mgmt_network_id,
+                 'fixed_ips': [{"subnet_id": svc_mgmt_ptg_subnet_id}],
+                 'port_security_enabled': True,
+                 'admin_state_up': True,
+                 #'security_groups': [{"id": sg_grp_id}],
+                 'mac_address': attributes.ATTR_NOT_SPECIFIED,
+                 'device_id': '',
+                 'device_owner': ''
+                }}
+        with admin_context.session.begin(subtransactions=True):
+            mgmt_port = self._core_plugin.create_port(
```

Comment:
I seem to have lost the context of why we need to crate port explicitly, instead of just calling create_pt? Port security is still enabled on this port.